python/django_bolt/concurrency.py

```
@@ -74,18 +75,58 @@ def _default_orm_workers() -> int:
+
+
+# Guards lazy construction. Check-then-create was safe while the only caller
+# ran on the event loop thread; user loading forces `request.user` from
+# arbitrary threads, where two racing callers would each build a pool and the
+# loser would leak its threads and their database connections — precisely the
+# accounting this pool exists to enforce.
+_orm_executor_lock = threading.Lock()
+
+
 def _get_orm_executor() -> concurrent.futures.ThreadPoolExecutor:
     global _orm_executor
```

Ruff (0.16.0)

[warning] Using the global statement to update `_orm_executor` is discouraged

(PLW0603)